app/controllers/api/v1/project_roles_controller.rb

```diff
@@ -10,9 +10,13 @@ def resource_name
   end
 
   def update
-    super do
-      UserAddedToProjectMailerWorker.perform_async(api_user.id, controlled_resource.resource.id, roles) if new_roles_present?(roles)
-    end
+    UserAddedToProjectMailerWorker.perform_async(acl_user_id, controlled_resource.resource.id, roles) if new_roles_present?(roles)
```

Comment:
re the yield order, create will run after the [save!](https://github.com/zooniverse/Panoptes/blob/7c0106054b20b4e13dc8e5f345784d8679b60e58/lib/json_api_controller/creatable_resource.rb#L17) but update will run before the [save!](https://github.com/zooniverse/Panoptes/blob/7c0106054b20b4e13dc8e5f345784d8679b60e58/lib/json_api_controller/updatable_resource.rb#L20). 

I'd move the update to be after super to make sure an update error blocks the email going out and leverage the yield block in create. Double check it's not the if statement stopping the code executing though a debug statement in the yielded block should run regardless.